docs/pipeline/enrichments/available-enrichments/custom-javascript-enrichment/writing/index.md

```diff
@@ -243,7 +243,31 @@ You might be tempted to update derived entities in a similar way by using `event
 
 ## Discarding the event
 
-Sometimes you don’t want the event to appear in your data warehouse or lake, e.g. because you suspect it comes from a bot and not a real user. In this case, you can `throw` an exception in your JavaScript code, which will send the event to [failed events](/docs/fundamentals/failed-events/index.md):
+Sometimes you don’t want the event to appear in your data warehouse or lake, e.g. because you suspect it comes from a bot and not a real user.
+
+Starting with Enrich 5.3.0, it is possible to drop an event by calling `event.drop()` in JavaScript enrichment code:
+
+```js
+const botPattern = /.*Googlebot.*/;
+
+function process(event) {
+  const useragent = event.getUseragent();
+
+  if (useragent !== null && botPattern.test(useragent)) {
+    event.drop();
+  }
+}
+```
+
+This mechanism can be used to drop not only good events, but also invalid events. The dropped events will not be sent to any stream or destination, thus lowering the infrastructure costs.
```

Comment:
Does the incomplete stream need to be switched on for this to work?

Reply:
No, it doesn't.